hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java

```diff
@@ -318,6 +326,8 @@ public void close() throws StorageContainerException {
     // It is ok if this operation takes a bit of time.
     // Close container is not expected to be instantaneous.
     compactDB();
+    // We must sync the DB after close operation
+    flushAndSyncDB();
```

Comment:
updateContainerData takes care to revert the state of the container in case of failure.
Won't that be a problem with flushDB? If flush fails, don't we want the container to remain Open?
Also, related question : what should be the relative order - sync first and then change state to close, both with the writeLock held

Same question applies to quasiClose as well.

Reply:
The recent commit moves sync before state change. Any failure in flush should now move to container to UNHEALTHY state in HddsDispatcher#dispatchRequest.